diff -up meshlab-snapshot-svn3524/meshlab/src/fgt/filter_func/filter_func.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/fgt/filter_func/filter_func.pro
--- meshlab-snapshot-svn3524/meshlab/src/fgt/filter_func/filter_func.pro.sharedlib	2009-05-20 08:27:47.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/fgt/filter_func/filter_func.pro	2010-01-26 17:22:25.000000000 -0800
@@ -7,7 +7,7 @@ SOURCES       = filter_func.cpp\ 
 
 TARGET        = filter_func
 
-INCLUDEPATH += ../../external/muparser_v130/include
+INCLUDEPATH += 
 
 # Note: we need static libs so when building muparser lib use 
 # ./configure --enable-shared=no
@@ -15,5 +15,5 @@ INCLUDEPATH += ../../external/muparser_v
 win32-msvc2005:LIBS += ../../external/lib/win32-msvc2005/muparser.lib
 win32-msvc2008:LIBS += ../../external/lib/win32-msvc2008/muparser.lib
 macx:LIBS            += ../../external/lib/macx/libmuparser.a
-linux-g++:LIBS       += ../../external/lib/linux-g++/libmuparser.a
+linux-g++:LIBS       += -lmuparser
 win32-g++:LIBS		 += ../../external/lib/win32-gcc/libmuparser.a
\ No newline at end of file
diff -up meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/filter_qhull.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/filter_qhull.pro
--- meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/filter_qhull.pro.sharedlib	2009-03-05 16:09:52.000000000 -0800
+++ meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/filter_qhull.pro	2010-01-26 17:22:25.000000000 -0800
@@ -12,29 +12,8 @@ SOURCES       = filter_qhull.cpp \ 
 	   	../../meshlab/filterparameter.cpp
 		
 
-HEADERS += 	../../external/qhull-2003.1/src/geom.h \
-		../../external/qhull-2003.1/src/io.h \
-		../../external/qhull-2003.1/src/mem.h \
-		../../external/qhull-2003.1/src/merge.h \
-		../../external/qhull-2003.1/src/poly.h \
-		../../external/qhull-2003.1/src/qhull.h \
-		../../external/qhull-2003.1/src/qset.h \
-		../../external/qhull-2003.1/src/stat.h \
-		../../external/qhull-2003.1/src/user.h 
+INCLUDEPATH += /usr/include/qhull
 
-SOURCES +=	../../external/qhull-2003.1/src/geom.c \
-		../../external/qhull-2003.1/src/geom2.c \
-		$$GLEWCODE \
-		../../external/qhull-2003.1/src/global.c \
-		../../external/qhull-2003.1/src/io.c \
-		../../external/qhull-2003.1/src/mem.c \
-		../../external/qhull-2003.1/src/merge.c \
-		../../external/qhull-2003.1/src/poly.c \
-		../../external/qhull-2003.1/src/poly2.c \
-		../../external/qhull-2003.1/src/qconvex.c \
-		../../external/qhull-2003.1/src/qhull.c \
-		../../external/qhull-2003.1/src/qset.c \
-		../../external/qhull-2003.1/src/stat.c \
-		../../external/qhull-2003.1/src/user.c 
+LIBS += -lqhull
 
 TARGET        = filter_qhull
\ No newline at end of file
diff -up meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/qhull_tools.h.sharedlib meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/qhull_tools.h
--- meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/qhull_tools.h.sharedlib	2009-05-30 09:08:03.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/fgt/filter_qhull/qhull_tools.h	2010-01-26 17:22:25.000000000 -0800
@@ -55,14 +55,14 @@ extern "C"
 #endif
 #include <stdio.h>
 #include <stdlib.h>
-#include "../../external/qhull-2003.1/src/qhull.h"
-#include "../../external/qhull-2003.1/src/mem.h"
-#include "../../external/qhull-2003.1/src/qset.h"
-#include "../../external/qhull-2003.1/src/geom.h"
-#include "../../external/qhull-2003.1/src/merge.h"
-#include "../../external/qhull-2003.1/src/poly.h"
-#include "../../external/qhull-2003.1/src/io.h"
-#include "../../external/qhull-2003.1/src/stat.h"
+#include <qhull/qhull.h>
+#include <qhull/mem.h>
+#include <qhull/qset.h>
+#include <qhull/geom.h>
+#include <qhull/merge.h>
+#include <qhull/poly.h>
+#include <qhull/io.h>
+#include <qhull/stat.h>
 #if defined(__cplusplus)
 }
 #endif
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlab/meshlab.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlab/meshlab.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlab/meshlab.pro.sharedlib	2009-07-06 09:45:12.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlab/meshlab.pro	2010-01-26 17:22:25.000000000 -0800
@@ -1,13 +1,11 @@
 VCGDIR  = ../../../vcglib
-GLEWDIR = ../external/glew-1.5.1
-GLEWCODE = $$GLEWDIR/src/glew.c
-DEFINES *= GLEW_STATIC
+LIBS += -lGLEW
 CONFIG += debug_and_release precompile_header
 # uncomment to try Eigen
 # DEFINES += VCG_USE_EIGEN
 # CONFIG += warn_off
 
-INCLUDEPATH  *= ../.. $$VCGDIR $$GLEWDIR/include
+INCLUDEPATH  *= ../.. $$VCGDIR
 DEPENDPATH  += $$VCGDIR  $$VCGDIR/vcg $$VCGDIR/wrap
 
 PRECOMPILED_HEADER = mainwindow.h
@@ -51,8 +49,7 @@ SOURCES        = main.cpp \
                  changetexturename.cpp \
                  stdpardialog.cpp \
                  $$VCGDIR/wrap/gui/trackball.cpp\
-                 $$VCGDIR/wrap/gui/trackmode.cpp \
-                 $$GLEWCODE
+                 $$VCGDIR/wrap/gui/trackmode.cpp
 
 FORMS          =    ui/layerDialog.ui \
                     ui/filterScriptDialog.ui \
@@ -102,9 +99,7 @@ mac:QMAKE_CXX=g++-4.2
 # macx-g++:QMAKE_CXXFLAGS_RELEASE -= -Os
 # macx-g++:QMAKE_CXXFLAGS_RELEASE += -O3
 
-DEFINES += GLEW_STATIC
-
-INCLUDEPATH += . .. ../../../vcglib $$GLEWDIR/include
+INCLUDEPATH += . .. ../../../vcglib
 CONFIG += stl
 
 # uncomment in your local copy only in emergency cases. 
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/edit_phototexturing/edit_phototexturing.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/edit_phototexturing/edit_phototexturing.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/edit_phototexturing/edit_phototexturing.pro.sharedlib	2009-05-18 08:12:15.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/edit_phototexturing/edit_phototexturing.pro	2010-01-26 17:22:25.000000000 -0800
@@ -30,7 +30,6 @@ SOURCES += 	edit_phototex_factory.cpp \
 			src/QuadTree/QuadTreeNode.cpp \
 			$$VCGDIR/wrap/gui/trackball.cpp\
 			$$VCGDIR/wrap/gui/trackmode.cpp \
-			$$GLEWDIR/src/glew.c \
 			../../meshlab/filterparameter.cpp \
 			../../meshlab/stdpardialog.cpp 
 
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/filter_isoparametrization.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/filter_isoparametrization.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/filter_isoparametrization.pro.sharedlib	2010-01-26 17:53:15.000000000 -0800
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/filter_isoparametrization.pro	2010-01-26 17:53:44.000000000 -0800
@@ -28,8 +28,7 @@ SOURCES       = ./filter_isoparametrizat
 				$$GLEWCODE
 				
 TARGET        = filter_isoparametrization
-INCLUDEPATH  += ./  \
-                ../../external/levmar-2.3/
+INCLUDEPATH  += ./
 
 win32-msvc2005:QMAKE_CXXFLAGS   +=  /openmp
 win32-msvc2008:QMAKE_CXXFLAGS   +=  /openmp
@@ -40,7 +39,7 @@ win32-msvc.net:LIBS	+= ../../external/li
 win32-msvc2005:LIBS	+= ../../external/lib/win32-msvc2005/levmar.lib
 win32-msvc2008:LIBS	+= ../../external/lib/win32-msvc2008/levmar.lib
 win32-g++:LIBS		+= -L../../external/lib/win32-gcc -llevmar
-linux-g++:LIBS		+= -L../../external/lib/win32-gcc -llevmar
+linux-g++:LIBS		+= -llevmar
 macx:LIBS += ../../external/lib/macx/liblevmar.a
 
 QMAKE_CXXFLAGS += -fpermissive
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h.sharedlib	2010-01-26 17:51:30.000000000 -0800
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/param_collapse.h	2010-01-26 17:51:56.000000000 -0800
@@ -12,7 +12,7 @@
 #include <local_parametrization.h>
 #include <mesh_operators.h>
 #include <vcg/space/color4.h>
-#include <lm.h>
+#include <levmar.h>
 #include <uv_grid.h>
 #include "opt_patch.h"
 #include "local_optimization.h"
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/parametrizator.h.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/parametrizator.h
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/parametrizator.h.sharedlib	2010-01-26 17:51:37.000000000 -0800
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/filter_isoparametrization/parametrizator.h	2010-01-26 17:52:11.000000000 -0800
@@ -30,7 +30,7 @@
 #include <vcg/space/color4.h>
 #include <dual_coord_optimization.h>
 #include <float.h>
-#include <lm.h>
+#include <levmar.h>
 #include <wrap/io_trimesh/export_ply.h>
 //#include <EquilaterizeMesh.h>
 
@@ -914,4 +914,4 @@ public:
 
 };
 
-#endif
\ No newline at end of file
+#endif
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_3ds/io_3ds.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_3ds/io_3ds.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_3ds/io_3ds.pro.sharedlib	2009-07-06 09:45:12.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_3ds/io_3ds.pro	2010-01-26 17:22:25.000000000 -0800
@@ -14,7 +14,7 @@ SOURCES       = 	meshio.cpp \
 
 TARGET        = io_3ds
 
-INCLUDEPATH += ../../external/lib3ds-1.3.0/
+INCLUDEPATH += /usr/include/lib3ds/
 
 # Notes on the paths of lib3ds files.
 # Once you have compiled the library put the lib files in a dir named lib/XXX/ 
@@ -25,7 +25,7 @@ win32-msvc.net:LIBS	+= ../../external/li
 win32-msvc2005:LIBS	+= ../../external/lib/win32-msvc2005/3ds.lib
 win32-msvc2008:LIBS	+= ../../external/lib/win32-msvc2008/3ds.lib
 win32-g++:LIBS		+= ../../external/lib/win32-gcc/lib3ds.a
-linux-g++:LIBS += ../../external/lib/linux-g++/lib3ds.a
+linux-g++:LIBS += -l3ds
 
 # unix:LIBS		+= -L../../../../code/lib/lib3ds-1.3.0/lib3ds/lib/unix -l3ds
 
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_epoch/io_epoch.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_epoch/io_epoch.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_epoch/io_epoch.pro.sharedlib	2009-09-01 01:33:26.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabplugins/io_epoch/io_epoch.pro	2010-01-26 17:22:25.000000000 -0800
@@ -31,12 +31,12 @@ QT           += xml
 win32-msvc2005:LIBS	   += ../../external/lib/win32-msvc2005/bz2.lib
 win32-msvc2008:LIBS	   += ../../external/lib/win32-msvc2008/bz2.lib
 win32-g++:LIBS	+= ../../external/lib/win32-gcc/libbz2.a
-linux-g++:LIBS	+= ../../external/lib/linux-g++/libbz2.a
+linux-g++:LIBS	+= -lbz2
 
 win32-msvc2005:INCLUDEPATH += ../../external/bzip2-1.0.5
 win32-msvc2008:INCLUDEPATH += ../../external/bzip2-1.0.5
 win32-g++:INCLUDEPATH  += ../../external/bzip2-1.0.5
-linux-g++:INCLUDEPATH	+= ../../external/bzip2-1.0.5
+linux-g++:INCLUDEPATH	+= 
 
 mac:LIBS   += -lbz2
 unix:LIBS += -lbz2
diff -up meshlab-snapshot-svn3524/meshlab/src/meshlabserver/meshlabserver.pro.sharedlib meshlab-snapshot-svn3524/meshlab/src/meshlabserver/meshlabserver.pro
--- meshlab-snapshot-svn3524/meshlab/src/meshlabserver/meshlabserver.pro.sharedlib	2009-05-20 07:51:45.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlabserver/meshlabserver.pro	2010-01-26 17:22:25.000000000 -0800
@@ -1,5 +1,4 @@
-GLEWDIR = ../external/glew-1.5.1
-GLEWCODE = $$GLEWDIR/src/glew.c
+LIBS += -lGLEW
 
 HEADERS        = ../meshlab/interfaces.h \
                  ../meshlab/meshmodel.h \
@@ -10,8 +9,7 @@ SOURCES        = mainserver.cpp \
                  ../meshlab/meshmodel.cpp \
                  ../meshlab/filterscript.cpp \
                  ../meshlab/filterparameter.cpp \
-                 ../meshlab/plugin_support.cpp \
-                 $$GLEWCODE
+                 ../meshlab/plugin_support.cpp
 
 # to add windows icon 
 RC_FILE = ../meshlab/meshlab.rc
@@ -22,11 +20,9 @@ QT           += xml opengl
 # the awful min/max macros of windows and the limits max
 win32:DEFINES += NOMINMAX
 
-DEFINES += GLEW_STATIC
-
-INCLUDEPATH += . .. ../../../vcglib $$GLEWDIR/include
+INCLUDEPATH += . .. ../../../vcglib
 CONFIG += stl 
 CONFIG += console
 
 # Mac specific Config required to avoid to make application bundles
-CONFIG -= app_bundle
\ No newline at end of file
+CONFIG -= app_bundle
diff -up meshlab-snapshot-svn3524/meshlab/src/shared.pri.sharedlib meshlab-snapshot-svn3524/meshlab/src/shared.pri
--- meshlab-snapshot-svn3524/meshlab/src/shared.pri.sharedlib	2009-07-07 08:12:05.000000000 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/shared.pri	2010-01-26 17:22:25.000000000 -0800
@@ -6,15 +6,14 @@ CONFIG       += plugin
 QT += xml
 
 VCGDIR  = ../../../../vcglib
-GLEWDIR = ../../external/glew-1.5.1
-GLEWCODE = $$GLEWDIR/src/glew.c
-DEFINES *= GLEW_STATIC
+GLEWCODE =
+LIBS += -lGLEW
 
 # uncomment to try Eigen
 # DEFINES += VCG_USE_EIGEN
 # CONFIG += warn_off
 
-INCLUDEPATH  *= ../.. $$VCGDIR $$GLEWDIR/include
+INCLUDEPATH  *= ../.. $$VCGDIR /usr/include/GL
 
 # Uncomment these if you want to experiment with newer gcc compilers
 # (here using the one provided with macports)
