diff -up meshlab-snapshot-svn3524/meshlab/src/meshlab/filterparameter.cpp.gcc45 meshlab-snapshot-svn3524/meshlab/src/meshlab/filterparameter.cpp
--- meshlab-snapshot-svn3524/meshlab/src/meshlab/filterparameter.cpp.gcc45	2010-09-10 19:04:04.955733001 -0700
+++ meshlab-snapshot-svn3524/meshlab/src/meshlab/filterparameter.cpp	2010-09-10 19:17:12.800733003 -0700
@@ -535,7 +535,7 @@ void RichParameterFactory::create( const
 
 	if(type=="RichEnum")
 	{
-		QStringList list = QStringList::QStringList();
+		QStringList list = QStringList();
 		unsigned int enum_card = np.attribute(QString("enum_cardinality")).toUInt();
 
 		for(int i=0;i<enum_card;++i)
@@ -582,4 +582,4 @@ void RichParameterFactory::create( const
 	}
 
 	assert(0); // we are trying to parse an unknown xml element
-}
\ No newline at end of file
+}
