diff -up meshlab-snapshot-svn3524/meshlab/src/meshlab/interfaces.h.plugin-path meshlab-snapshot-svn3524/meshlab/src/meshlab/interfaces.h
--- meshlab-snapshot-svn3524/meshlab/src/meshlab/interfaces.h.plugin-path	2010-02-02 18:38:30.000000000 -0800
+++ meshlab-snapshot-svn3524/meshlab/src/meshlab/interfaces.h	2010-02-02 18:39:29.000000000 -0800
@@ -199,7 +199,7 @@ public:
 	
 	static QString getPluginDirPath()
 	{
-		QDir pluginsDir(getBaseDirPath());
+		QDir pluginsDir(QString(PLUGIN_DIR));
 		if(!pluginsDir.exists("plugins"))
 				QMessageBox::warning(0,"Meshlab Initialization","Serious error. Unable to find the plugins directory.");
 		
